### ISSUE TITLE
FileSaver works for me just for .txt

### ISSUE THREAD

**Issue by migorman** [eligrey/FileSaver.js#183]
Hi I'm using FileSaver to save my files with different extension but it woks just with .txt
  var blob = new Blob([data], { type: contentType });
contentType is a variable here
    FileSaver.saveAs(blob, fileName);
for .pdf for example i'm getting failed to load pdf document is there any solution for this !!

**Comment by jimmywarting** [comment]
It shouldn't matter what content type it's if you are just downloading it.
The content type is only helpful for the browser to render or execute a file correctly.

could you provide some more info like what OS & browser you are using?
also is the `data` just a simple plain text or is it actually the content of a pdf?

for example this wouldn't work cuz it's just plain text and would need to be converted to a pdf format

     saveAs(new Blob(['hello world'], {type: application/pdf}), 'sample.pdf')

**Comment by migorman** [comment]
Hi @jimmywarting  i already found a solution for this,thanks for your reply :)

**Comment by jimmywarting** [comment]
then we can close this issue :)

**Comment by migorman** [comment]
yeah thanks :)

2015-12-01 13:49 GMT+00:00 Jimmy Karl Roland Wärting <
notifications@github.com>:

> then we can close this issue :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eligrey/FileSaver.js/issues/183#issuecomment-160975041>
> .
>



-- 
Mounir IGORMAN
Ingénieur Etudes & Développement
Phone : +212617612986

**Comment by theonlysteve** [comment]
I'm having a similar issues saving a PDF.

@migorman, do you mind sharing your solution?

Thanks!

**Comment by jimmywarting** [comment]
@theonlysteve is [this](http://stackoverflow.com/questions/18191893/generate-pdf-from-html-in-div-using-javascript) what you want?

**Comment by Gonzo17** [comment]
So, I have the same issue. PDF is not working. :/ Does anyone know the solution?

**Comment by jimmywarting** [comment]
@Gonzo17 can you tell why my car isn't working? No cuz you haven't seen it.

You need to show us some code for us to help you. How do you get the pdf content?

Probably the main cause is that your blob is probably constructed with a string or a binary string. It's nothing wrong with FileSaver but rather who you get the pdf binaries.

**Comment by Gonzo17** [comment]
@jimmywarting For sure you don't know what's wrong without any information. :) I understand your point of view. I'm new to JS and I was pretty frustrated when I saw that someone found a solution but didn't share it.

Having said that, I'm happy that I solved the issue myself. Indeed it was a problem with the http response. The body is sent as 'Text' by default. I had to add a 'responseType', in this case 'ArrayBuffer'. My example:

`this.http.get(url, {responseType: ResponseContentType.ArrayBuffer})`

The following discussions helped me to figure this out:
http://stackoverflow.com/questions/28093736/failed-to-load-pdf-document-angular-js-blob
http://stackoverflow.com/questions/35138424/how-do-i-download-a-file-with-angular2
